collect_app/src/main/java/org/odk/collect/android/tasks/InstanceSyncTask.java

```diff
@@ -107,7 +107,9 @@ protected String doInBackground(Void... params) {
                     while (instanceCursor.moveToNext()) {
                         String instanceFilename = instanceCursor.getString(
                                 instanceCursor.getColumnIndex(InstanceColumns.INSTANCE_FILE_PATH));
-                        if (candidateInstances.contains(instanceFilename)) {
+                        String instanceStatus = instanceCursor.getString(
```

Comment:
@grzesiek2010 how about we add the status restriction in the query? That way we will get less records from the db and we don't need to do iteration here.